New tests

whisperthorn opened the pull request.

whisperthorn committed:
New tests

okryukova commented on src/test/java/POM/MainPage.java:
```diff
@@ -0,0 +1,55 @@
+package POM;
```
⛔️Нужно исправить. Необходимо исправить наименование пакетов в соответствии с правилами нейминга. Пакеты — допустимы только строчные буквы. В случае, если слов несколько, их следует разделять точкой. Префикс должен быть одним из доменных имен верхнего уровня, таких как com, ru и т.д. Наименование пакета должно отражать суть содержимого. https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html

okryukova commented on src/test/java/POM/MainPage.java:
```diff
@@ -0,0 +1,55 @@
+
+    public boolean isAnswerFound(String questionAnswer){
+        try {
+            driver.findElement(By.xpath(String.format(".//p[text()='%s']", questionAnswer)));
```
⛔️Нужно исправить. Локаторы необходимо вынести в поля класса. Так их можно переиспользовать в коде. Поправь, пожалуйста, этот момент во всем коде

okryukova commented on src/test/java/POM/MainPage.java:
```diff
@@ -0,0 +1,55 @@
+
+public class MainPage {
+    private static final String MAIN_PAGE_URL = "https://qa-scooter.praktikum-services.ru/";
+    private static final By SCOOTER_LOGO = By.className("Header_LogoScooter__3lsAR");
```
✅Отлично. Молодец, что используешь относительные , а не полные локаторы. Такой код меньше подвержен изменениям.

okryukova commented on src/test/java/POM/OrderScooterPage.java:
```diff
@@ -0,0 +1,208 @@
+
+    public static final String[] RENT_PERIOD_STRING = new String[]{"сутки", "двое суток", "трое суток", "четверо суток", "пятеро суток", "шестеро суток", "семеро суток"};
+    private static final String[] MONTH_NAMES = new String[]{"января", "февраля", "марта", "апреля", "мая", "июня", "июля", "августа", "сентября", "октября", "ноября", "декабря"};
+    private static final String MAIN_PAGE_URL = "https://qa-scooter.praktikum-services.ru/";
```
⚠️Можно улучшить.  Константа присутствует в каждом классе. Можно ее вынести в отдельный класс, сделать публичной и использовать во всем проекте

okryukova commented on src/test/java/POM/OrderScooterPage.java:
```diff
@@ -0,0 +1,208 @@
+    public static final String[] RENT_PERIOD_STRING = new String[]{"сутки", "двое суток", "трое суток", "четверо суток", "пятеро суток", "шестеро суток", "семеро суток"};
+    private static final String[] MONTH_NAMES = new String[]{"января", "февраля", "марта", "апреля", "мая", "июня", "июля", "августа", "сентября", "октября", "ноября", "декабря"};
+    private static final String MAIN_PAGE_URL = "https://qa-scooter.praktikum-services.ru/";
+    private static final By UPPER_ORDER_BUTTON = By.xpath(".//div[@class='Header_Nav__AGCXC']//button[@class='Button_Button__ra12g']");
```
⛔️Нужно исправить. Кнопки заказа находятся на Main Page.  Их нужно перенести в правильный класс. А так же локатор для куки

okryukova commented on src/test/java/QATextTest.java:
```diff
@@ -0,0 +1,53 @@
+    @Parameterized.Parameters
+    public static Object[][] getTextData() {
+        return new Object[][]{
+                {"Сколько это стоит? И как оплатить?", "Сутки — 400 рублей. Оплата курьеру — наличными или картой."},
```
⚠️Можно улучшить. Тестовые данные  должны содержаться либо в константах тестового класса, либо в отдельном файле

okryukova commented on src/test/java/QATextTest.java:
```diff
@@ -0,0 +1,53 @@
+
+    @Before
+    public void before() {
+        driver = new FirefoxDriver();
```
⚠️Можно улучшить. Согласно заданию, проводить тестирование необходимо в двух браузерах. Необходимо реализовать возможность выбора браузера. Выбор браузера можно осуществлять через системные свойства System.getProperties().

okryukova replied:
⚠️Можно улучшить. Блок открытия браузера одинаковый для всех тестовых классов. Целесообразно вынести его в базовый или утилитный класс.
⚠️Можно улучшить. Браузер, как правило, запускают в полноэкранном режиме (maximized) или определенном размере окна.

okryukova commented on src/test/java/QATextTest.java:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    @Test
+    public void checkAnswerText() {
```
⚠️Можно улучшить.  В названии метода необходимо использовать слово Test.  Проверим соблюдается ли данное правило для всех созданных тестов.

okryukova commented on src/test/java/QATextTest.java:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    @After
+    public void tearDown() {
```
⚠️Можно улучшить. Блок закрытия браузера одинаковый для всех тестовых классов. Целесообразно вынести его в базовый или утилитный класс.

okryukova commented on src/test/java/LogoUrlTest.java:
```diff
@@ -0,0 +1,46 @@
+import org.openqa.selenium.support.ui.WebDriverWait;
+import java.time.Duration;
+
+public class LogoUrlTest {
```
✅Отлично. Молодец, что реализовал доп. тесты

okryukova commented on src/test/java/POM/MainPage.java:
```diff
@@ -0,0 +1,55 @@
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.JavascriptExecutor;
+
+public class MainPage {
```
⚠️Можно улучшить. Для РО обычно создают пакет в src/main/java. Будет круто, если ты перенесешь туда свои РО

whisperthorn committed:
Перенес pom-ы в scr\main\java
Переименовал название пакета согласно правилам нейминга
Перенес локаторы и методы кнопок Заказать и принять cookie в MainPage pom
Добавил test в названия тестов в которых его не было
Добавил неявное ожидание для тестов
Добавил явное ожидание для подгрузки станций метро
Добавил ожидание кликательности нижней кнопки Заказать

whisperthorn committed:
Перенес pom-ы в scr\main\java
Переименовал название пакета согласно правилам нейминга
Перенес локаторы и методы кнопок Заказать и принять cookie в MainPage pom
Вынес статическую часть xpath динамических локаторов в поля класса
Добавил test в названия тестов в которых его не было
Добавил неявное ожидание для тестов
Добавил явное ожидание для подгрузки станций метро
Добавил ожидание кликательности нижней кнопки Заказать

whisperthorn committed:
Merge remote-tracking branch 'origin/develop' into develop

whisperthorn commented on src/main/java/ru/praktikum/scooter/OrderScooterPage.java:
```diff
@@ -30,31 +26,15 @@ public class OrderScooterPage {
     private static final By FINISH_ORDER_BUTTON = By.xpath(".//button[@class='Button_Button__ra12g Button_Middle__1CSJM' and text()='Заказать']");
     private static final By CONFIRM_ORDER_BUTTON = By.xpath(".//button[@class='Button_Button__ra12g Button_Middle__1CSJM' and text()='Да']");
     private static final By SUCCESSFUL_ORDER_TEXT = By.xpath(".//div[@class='Order_ModalHeader__3FDaJ' and text()='Заказ оформлен']");
+    public static final String XPATH_SEARCH_RENT_PERIOD = ".//div[text()='%s']";
```
Так же вынес статическую xpath часть динамических локаторов в поля класса

okryukova commented on src/test/java/QATextTest.java:
```diff
@@ -0,0 +1,57 @@
+        MainPage objMainPage = new MainPage(driver);
+        objMainPage.clickQuestion(questionHeader);
+        Assert.assertTrue("Текст ответа не найден", objMainPage.isAnswerFound(questionAnswer));
+        Assert.assertTrue("Текст ответа не отображается", objMainPage.isAnswerVisible(questionAnswer));
```
⛔️Нужно исправить. Часть тестов падает на этой проверке. Тесты должны проходить. 
В тесте будет достаточно одно ассерта с проверкой текста ответа

whisperthorn committed:
Добавил явное ожидание наличия FQA элементов на странице, чтобы элемент успевал загрузится.
Добавил явное ожидание клибательности FQA элементов, на случай если картинка самоката не успевает исчезнуть и блокирует FQA элементы.
